### PR TITLE
Addd examples on how to connect with Sigv4 for AOS and AOSS

### LINKED ISSUE #2925: [DOC] Add Sigv4 signing details to all client setup documentation
To ensure that Opensearch can be easily used in managed service and serverless environments, it is necessary to update the client instantiation documentation to include instructions on how to implement Sigv4 signing.

These instructions should be clear and easy to follow, with examples provided where possible. They should also include any relevant configuration options or code snippets that may be necessary to ensure successful implementation of Sigv4 signing.

By updating the client instantiation documentation to include these instructions, we can make it easier for developers to use Opensearch in a wider range of environments, which will ultimately help to grow the Opensearch user community and improve the overall quality of the project.

### PR DESCRIPTION
**Pull request by harshavamsi**
### Description
This PR adds examples for Sigv4 signing to all clients for both the Amazon OpenSearch service and the Amazon OpenSearch Serverless Service

### Issues Resolved
Closes #2925 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _clients/go.md [hunk 1]
```diff
@@ -55,6 +55,126 @@ client, err := opensearch.NewClient(opensearch.Config{
 ```
 {% include copy.html %}
 
+## Connecting to the Amazon OpenSearch Service
```
Comment (kolchfa-aws) [inline _clients/go.md:58]:
```suggestion
## Connecting to Amazon OpenSearch Service
```

File: _clients/go.md [hunk 2]
```diff
@@ -55,6 +55,126 @@ client, err := opensearch.NewClient(opensearch.Config{
...
+			SecretAccessKey: secretAccessKey,
+			SessionToken:    token,
+		}
+		return *c, nil
+	}
+}
+```
+{% include copy.html %}
+
+## Connecting to the Amazon OpenSearch Serverless Service
```
Comment (kolchfa-aws) [inline _clients/go.md:118]:
```suggestion
## Connecting to Amazon OpenSearch Serverless
```

File: _clients/java.md [hunk 3]
```diff
@@ -150,6 +150,51 @@ public class OpenSearchClientExample {
 ```
 {% include copy.html %}
 
+## Connecting to the Amazon OpenSearch Service
```
Comment (kolchfa-aws) [inline _clients/java.md:153]:
```suggestion
## Connecting to Amazon OpenSearch Service
```

File: _clients/java.md [hunk 4]
```diff
@@ -150,6 +150,51 @@ public class OpenSearchClientExample {
...
+);
+
+InfoResponse info = client.info();
+System.out.println(info.version().distribution() + ": " + info.version().number());
+
+httpClient.close();
+```
+{% include copy.html %}
+
+## Connecting to the Amazon OpenSearch Serverless Service
```
Comment (kolchfa-aws) [inline _clients/java.md:175]:
```suggestion
## Connecting to Amazon OpenSearch Serverless 
```

File: _clients/php.md [hunk 5]
```diff
@@ -44,7 +44,50 @@ $client = (new \OpenSearch\ClientBuilder())
...
+    ->setSigV4CredentialProvider([
+      'key' => 'awskeyid',
+      'secret' => 'awssecretkey',
+    ])
+    
+    ->build();
+```
+{% include copy.html %} 
+
+## Connecting to Amazon OpenSearch Serverless Service
```
Comment (kolchfa-aws) [inline _clients/php.md:70]:
```suggestion
## Connecting to Amazon OpenSearch Serverless 
```

**Comment by harshavamsi** [comment]
Hi @kolchfa-aws I've updated the docs with the suggestions, thanks!

**Review by carolxob — APPROVED** [review]

**Pull request merged**